playwright/fixtures/index.ts

```diff
@@ -35,113 +44,242 @@ export type Fixtures = {
+      const customTypeName = "Custom Type " + generateRandomId();
+      await customTypesTablePage.createTypeDialog.createType(customTypeName);
 
-    const sliceName = "Slice" + generateRandomId();
-    await slicesListPage.createSliceDialog.createSlice(sliceName);
+      await use({ name: customTypeName });
+    },
+    slice: async ({ slicesListPage }, use) => {
+      await slicesListPage.goto();
+      await expect(slicesListPage.breadcrumbLabel).toBeVisible();
+      await slicesListPage.openCreateDialog();
 
-    await use({ name: sliceName });
-  },
+      const sliceName = "Slice" + generateRandomId();
+      await slicesListPage.createSliceDialog.createSlice(sliceName);
 
-  /**
-   * Page object override
-   */
-  page: async ({ browser }, use) => {
-    // Onboard user in Local Storage
-    const context = await browser.newContext({
-      storageState: {
-        cookies: [],
-        origins: [
-          {
-            origin: config.use.baseURL,
-            localStorage: [
-              {
-                name: "persist:root",
-                value: JSON.stringify({
-                  userContext: {
-                    userReview: {
-                      onboarding: true,
-                      advancedRepository: true,
+      await use({ name: sliceName });
+    },
+
+    /**
+     * Page
+     */
+    page: async ({ browser }, use) => {
+      // Onboard user in Local Storage by default
+      let context = await browser.newContext({
+        storageState: {
+          cookies: [],
+          origins: [
+            {
+              origin: config.use.baseURL,
+              localStorage: [
+                {
+                  name: "persist:root",
+                  value: JSON.stringify({
+                    userContext: {
+                      userReview: {
+                        onboarding: true,
+                        advancedRepository: true,
+                      },
+                      updatesViewed: {
+                        latest: null,
+                        latestNonBreaking: null,
+                      },
+                      hasSeenChangesToolTip: true,
+                      hasSeenSimulatorToolTip: true,
+                      hasSeenTutorialsToolTip: true,
+                      authStatus: "unknown",
+                      lastSyncChange: null,
                     },
-                    updatesViewed: { latest: null, latestNonBreaking: null },
-                    hasSeenChangesToolTip: true,
-                    hasSeenSimulatorToolTip: true,
-                    hasSeenTutorialsToolTip: true,
-                    authStatus: "unknown",
-                    lastSyncChange: null,
-                  },
-                }),
-              },
+                  }),
+                },
+                {
+                  name: "slice-machine_isInAppGuideOpen",
+                  value: "false",
+                },
+              ],
+            },
+          ],
+        },
```

Comment:
Should we be initializing pages with default state? In an ideal test (which we don't always need to achieve, to be clear), we would simulate what a user would do to get to that state. For example, if we want to test a page when a user has already onboarded, we would open a fresh page and close the onboarding dialog (or whatever the action is).

Doing it that way tests the full onboarding flow and expected states. It also means we can change implementation details in SM UI, like the property names used in localStorage, without modifying our tests.

Reply:
Discussed together 🙂